### PR TITLE
Add Hex to RZ conversion examples for blocks and reactors

### PR DESCRIPTION
**Pull request by ntouran**
Example of conversion of a CR with fuel driver to 1-D.

### CODE REVIEW

**Comment by ntouran** [comment]
Images this time are, for the block: 

image


and for the reactor: 

image

image


schmancy.

File: armi/reactor/converters/geometryConverters.py [hunk 1]
```diff
@@ -1100,8 +1111,8 @@ def plotConvertedReactor(self):
                     ax.add_patch(blockPatch)
                     innerAxial = outerAxial
                 innerRadius = outerRadius
-            plt.title(
-                "{} Core Map of from {} to {} revolutions".format(
+            ax.set_title(
+                    "{} Core Map of from {} to {:.4f} revolutions".format(
```
Comment (jakehader) [inline armi/reactor/converters/geometryConverters.py:1115]: Remove `of`

Comment (ntouran) [reply armi/reactor/converters/geometryConverters.py:1115]: oh yeah nice. thanks

File: doc/gallery-src/analysis/run_hexReactorToRZ.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+"""
+Hex reactor to RZ geometry conversion
+===================================
+This shows how an entire reactor specified in full hex detail can be
+automatically converted to an equivalent 2-D or 3-D RZ case.
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexReactorToRZ.py:5]: "equivalent" might be a bit strong. "representative?"

File: doc/gallery-src/analysis/run_hexBlockToRZConversion.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+armi.configure()
+
+o, r = test_reactors.loadTestReactor()
+
+bFuel = o.r.core.getBlocks(Flags.FUEL)[0]
+bControl = o.r.core.getBlocks(Flags.CONTROL)[0]
+converter = blockConverters.HexComponentsToCylConverter(
+    bControl, bFuel, numExternalRings=1
+)
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexBlockToRZConversion.py:39]: This is also pretty Stuff™. What is it doing? What are its arguments? What assumptions does it make about the internal geometry of the Block? What does `numExternalBlocks` mean? Which block is being converted, and why do I need to pass two blocks?

File: doc/gallery-src/analysis/run_hexBlockToRZConversion.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+from armi.reactor.flags import Flags
+from armi.reactor.converters import blockConverters
+import armi
+
+armi.configure()
+
+o, r = test_reactors.loadTestReactor()
+
+bFuel = o.r.core.getBlocks(Flags.FUEL)[0]
+bControl = o.r.core.getBlocks(Flags.CONTROL)[0]
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexBlockToRZConversion.py:36]: Get rid of `o`.

File: doc/gallery-src/analysis/run_hexBlockToRZConversion.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+from armi.reactor.tests import test_reactors
+from armi.reactor.flags import Flags
+from armi.reactor.converters import blockConverters
+import armi
+
+armi.configure()
+
+o, r = test_reactors.loadTestReactor()
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexBlockToRZConversion.py:33]: `_, r = test_reactors.loadTestReactor()` no one should have to know or think about `Operator` to understand this example.

File: doc/gallery-src/analysis/run_hexReactorToRZ.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+plotting.plotFaceMap(r.core, data=kgFis, labelFmt="{:.1e}")
+
+converterSettings = {
+    "radialConversionType": "Ring Compositions",
+    "axialConversionType": "Axial Coordinates",
+    "uniformThetaMesh": True,
+    "thetaBins": 1,
+    "axialMesh": [50, 100, 150, 175],
+    "thetaMesh": [2 * math.pi],
+}
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexReactorToRZ.py:32]: I can't find any documentation for this.

File: doc/gallery-src/analysis/run_hexReactorToRZ.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+converter = geometryConverters.HexToRZConverter(
+    o.cs, converterSettings
+)
+converter.convert(r)
```
Comment (youngmit) [inline doc/gallery-src/analysis/run_hexReactorToRZ.py:37]: What just happened? Did the `r` get converted in place? Did this function return a new, converted Reactor? Is the original `r` _gone_?

Comment (ntouran) [reply doc/gallery-src/analysis/run_hexReactorToRZ.py:37]: changed to not ditch the new reactor.

File: armi/reactor/converters/blockConverters.py [hunk 8]
```diff
@@ -402,13 +402,10 @@ def _addDriverFuelRings(self):
             mainComponent=fuel,
         )
 
-    def plotConvertedBlock(self):
+    def plotConvertedBlock(self, fName=None):
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/reactor/converters/blockConverters.py:405]: nope this is the only place that can plot a RZ converted block. classic tradeoff between putting the plotters by the thing they can plot or all together in utils.plotting here.

File: doc/gallery-src/analysis/run_hexReactorToRZ.py (hunk 2 above)
Comment (ntouran) [reply doc/gallery-src/analysis/run_hexReactorToRZ.py:5]: updated to say conserved mass

File: doc/gallery-src/analysis/run_hexBlockToRZConversion.py (hunk 3 above)
Comment (ntouran) [reply doc/gallery-src/analysis/run_hexBlockToRZConversion.py:39]: Added the keyword arguments and a hyperlink to the api docs at the top.

File: doc/gallery-src/analysis/run_hexBlockToRZConversion.py (hunk 5 above)
Comment (ntouran) [reply doc/gallery-src/analysis/run_hexBlockToRZConversion.py:33]: I made it `_o` soas to be slightly less mysterious.

File: doc/gallery-src/analysis/run_hexReactorToRZ.py (hunk 6 above)
Comment (ntouran) [reply doc/gallery-src/analysis/run_hexReactorToRZ.py:32]: ok I added some.

**Pull request merged**